### ISSUE TITLE
unnecessary_parenthesis with new/const

### ISSUE THREAD

**Issue by gabrc** [dart-lang/linter#1519]
The lint should not report an issue with the first line of the code below:
```
var result = !(const Foo()).bar();  // unnecessary_parenthesis reports issue
var result = !const Foo().bar();  // parenthesis removed, but urgh!
```

**Comment by srawlins** [comment]
The linter at head does not report lint on the first line ([test case on line 50](https://github.com/dart-lang/linter/blob/master/test/rules/unnecessary_parenthesis.dart#L50)). What version of the linter are you using?

**Comment by gabrc** [comment]
I'm on 0.1.82. It looks like it has been fixed at 0.1.83.

**Comment by pq** [comment]
Awesome.  Thanks for checking. I'll optimistically close this one.  Feel free to re-open if needed.  👍
